Orange/tests/test_table.py

```diff
@@ -2779,6 +2789,15 @@ def test_shared(self):
         self.assertEqual(4, call_cv.call_count)
         self.assertEqual(1, call_shared.call_count)
 
+        # test shared with row indices
+        new_table = Table.from_table(d1, self.iris, row_indices=[0, 1, 2])
+        np.testing.assert_array_equal(
+            new_table.X,
+            [[5.1, 5.1, 5.1, 5.1],
+             [4.9, 4.9, 4.9, 4.9],
+             [4.7, 4.7, 4.7, 4.7]]
+        )
+
```

Comment:
The new test belongs somewhere else. The tests in this class test if caching is working properly.